sw/device/tests/crypto/cryptotest/firmware/kmac_sca.c

```diff
@@ -0,0 +1,672 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/tests/crypto/cryptotest/firmware/kmac_sca.h"
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/dif/dif_kmac.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/ottf_test_config.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf.h"
+#include "sw/device/lib/ujson/ujson.h"
+#include "sw/device/sca/lib/prng.h"
+#include "sw/device/sca/lib/sca.h"
+#include "sw/device/tests/crypto/cryptotest/json/kmac_sca_commands.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "kmac_regs.h"
+
+enum {
+  /**
+   * Key length in bytes.
+   */
+  kKeyLength = 16,
+  /**
+   * Message length in bytes.
+   */
+  kMessageLength = 16,
+  /**
+   * Digest length in 32-bit words.
+   */
+  kDigestLength = 8,
+  /**
+   * Number of cycles (at `kClockFreqCpuHz`) that Ibex should sleep to minimize
+   * noise during SHA3 operations. Caution: This number should be chosen to
+   * provide enough time. Otherwise, Ibex might wake up while SHA3 is still busy
+   * and disturb the capture. Currently, we use a start trigger delay of 320
+   * clock cycles and the scope captures 160 clock cycles at kClockFreqCpuHz
+   * (3200 samples). On the scope side, an offset of 395 clock cycles (7900
+   * samples) is used to ignore 1) the start trigger delay for the PROCESS
+   * command, and 2) the absorb phase for the fixed prefix.
+   */
+  kIbexSha3SleepCycles = 1180,
```

Comment:
Also here, I would remove the samples numbers (see AES and SHA3).

Reply:
Done